### PR TITLE
Add Overview section

### LINKED ISSUE #1073: Add an Overview on the ECK website doc
Let's add a short Overview section, with a brief description of what the product is about and what kind of info users can find in the ECK doc.

### PR DESCRIPTION
**Pull request by alaudazzi**
Closes #1073

- Drafted the Overview section
- Updated the index.asciidoc to include the new file

Once this PR is merged, I would remove the following text from the Quickstart. 

> With Elastic Cloud on Kubernetes (ECK) you can extend the basic Kubernetes orchestration capabilities to easily deploy, secure, upgrade your Elasticsearch cluster, and much more.

### CODE REVIEW

File: docs/overview.asciidoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+== Overview
+
+Built on the Kubernetes Operator pattern, Elastic Cloud on Kubernetes (ECK) extends the basic Kubernetes orchestration capabilities to support the setup and management of Elasticsearch and Kibana on Kubernetes.
+
+With Elastic Cloud on Kubernetes you can streamline all those critical operations, such as:
+
+. Managing and monitoring multiple clusters
+. Scaling cluster capacity up and down
+. Changing cluster configuration
+. Dynamically scaling local storage (includes Elastic Local Volume, a local storage driver)
```
Comment (sebgl) [inline docs/overview.asciidoc]: Let's remove that one for now. We discussed how we should probably not advertise it too much until we commit to make it production-ready.

File: docs/overview.asciidoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+. Scheduling backups
```
Comment (sebgl) [inline docs/overview.asciidoc]: I think we can add:

```
. Securing clusters with TLS certificates
. Setting up hot-warm-cold architectures with availability zone awareness
```

File: docs/overview.asciidoc (hunk 1 above)
Comment (alaudazzi) [reply docs/overview.asciidoc]: Done.

File: docs/overview.asciidoc (hunk 2 above)
Comment (alaudazzi) [reply docs/overview.asciidoc]: Done.

**Comment by alaudazzi** [comment]
Thank you for your feedback @sebgl!

**Comment by alaudazzi** [comment]
Do we want the Overview appear for 0.8 as well?

**Comment by sebgl** [comment]
@alaudazzi Yes I think it's fine to also have in 0.8!

**Pull request merged**